x/stakeibc/keeper/icqcallbacks_withdrawal_balance.go

```diff
@@ -130,18 +119,49 @@ func WithdrawalBalanceCallback(k Keeper, ctx sdk.Context, args []byte, query icq
 	// Send the transaction through SubmitTx
 	_, err = k.SubmitTxsStrideEpoch(ctx, hostZone.ConnectionId, msgs, *withdrawalAccount, ICACallbackID_Reinvest, marshalledCallbackArgs)
 	if err != nil {
-		errMsg := fmt.Sprintf("Failed to SubmitTxs for %s - %s, Messages: %v | err: %s", hostZone.ChainId, hostZone.ConnectionId, msgs, err.Error())
-		k.Logger(ctx).Error(errMsg)
-		return sdkerrors.Wrapf(types.ErrICATxFailed, errMsg)
+		return sdkerrors.Wrapf(types.ErrICATxFailed, "Failed to SubmitTxs, Messages: %v, err: %s", msgs, err.Error())
 	}
 
 	ctx.EventManager().EmitEvent(
 		sdk.NewEvent(
 			sdk.EventTypeMessage,
 			sdk.NewAttribute("hostZone", hostZone.ChainId),
-			sdk.NewAttribute("totalWithdrawalBalance", withdrawalBalanceCoin.Amount.String()),
+			sdk.NewAttribute("totalWithdrawalBalance", withdrawalBalanceAmount.String()),
 		),
 	)
 
 	return nil
 }
+
+// Helper function to unmarshal a Balance query response across SDK versions
+// Before SDK v46, the query response returned a sdk.Coin type. SDK v46 returns an int type
+func UnmarshalAmountFromBalanceQuery(cdc codec.BinaryCodec, queryResponseBz []byte) (amount sdkmath.Int, err error) {
+	// An nil response would indicate some sort of error
+	if queryResponseBz == nil {
+		return sdkmath.Int{}, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "query response is nil")
+	}
+
+	// If the query response is empty, that means the account was never registed (and thus has a 0 balance)
+	if len(queryResponseBz) == 0 {
+		return sdkmath.ZeroInt(), nil
+	}
+
+	// First attempt to unmarshal as an Int (for SDK v46+)
```

Comment:
```suggestion
	// First attempt to unmarshal as an Int (for SDK v46+)
	// If the result was serialized as a `Coin` type, it should contain a string (representing the denom)
	// which will cause the unmarshalling to throw an error
```

Reply:
bit pedantic but we're making some assumptions here so I think it's better to explicitly list them

Reply:
good call!